internal/controllers/provisioningrequest_clusterconfig.go

```diff
@@ -32,47 +32,79 @@ func (t *provisioningRequestReconcilerTask) handleClusterPolicyConfiguration(ctx
 		return false, fmt.Errorf("status.clusterDetails is empty")
 	}
 
+	// List all the root policies in ztp-<ctNamespace>.
+	rootPolicies := &policiesv1.PolicyList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(fmt.Sprintf("ztp-%s", t.ctDetails.namespace)),
+	}
+	if err := t.client.List(ctx, rootPolicies, listOpts...); err != nil {
+		return false, fmt.Errorf("failed to list root Policies: %w", err)
+	}
+
+	// Derive the expected child policies set from root policies annotated with the
+	// ClusterTemplate reference "<name>.<version>".
+	// - If the annotation is absent, or the policy does not match the current ClusterTemplate,
+	//   we do not derive an expected set and consider the current present child policies only.
+	// - Store each policy's intended remediation to drive requeue/timeout decisions.
+	expectedChildPolicies := map[string]string{}
+	clusterTemplateRef := fmt.Sprintf("%s.%s", t.object.Spec.TemplateName, t.object.Spec.TemplateVersion)
+	for _, rootPolicy := range rootPolicies.Items {
+		if ctlrutils.RootPolicyMatchesClusterTemplate(rootPolicy.GetAnnotations(), clusterTemplateRef) {
+			t.logger.InfoContext(ctx, fmt.Sprintf("Detected matching root policy (%s in %s) for ClusterTemplate (%s)",
```

Comment:
I wonder if it might make more sense to use `DebugContext` instead of `InfoContext` here, since this looks more like debug-level information.